digits.go

```diff
@@ -6,41 +6,44 @@ type Expression struct {
 	sign       bool
 	nonSigFigs string
 	sigFigs    string
+	stringOut  string
 	head       string
 	core       string
 	tail       string
 }
 
-func New(p Precision, v string, g rune, d Decimals) (*Expression, error) {
-	if p != Exact && !(p > Trillions && p < Trillionth) {
+func New(significantPrecision Precision, value string, groupSeparator rune, decimalPrecision Decimals) (*Expression, error) {
+	if significantPrecision != Exact && !(significantPrecision > Trillions && significantPrecision < Trillionth) {
 		return nil, fmt.Errorf("precision out of bounds")
 	}
 	ret := Expression{}
-	value, err := lowPrecisionTruncate(p, v, d)
+	trunc, err := lowPrecisionTruncate(significantPrecision, value, decimalPrecision)
 	if err != nil {
 		return nil, err
 	}
-	ret.sign = value.Signbit()
-	sigFigs, err := computeSigFigs(p, v, g, d)
+	ret.sign = trunc.Signbit()
+	sigFigs, err := computeSigFigs(significantPrecision, value, groupSeparator, decimalPrecision)
 	if err != nil {
 		return nil, err
 	}
 	ret.sigFigs = sigFigs
-	nonSigFigs, err := computeNonSigFigs(p, v, d)
+	nonSigFigs, err := computeNonSigFigs(significantPrecision, value, decimalPrecision)
 	if err != nil {
 		return nil, err
 	}
 	ret.nonSigFigs = nonSigFigs
-	ret.head = computeHead(value)
-	core, err := computeCore(p, v, g, d)
+	stringOut, err := computeString(significantPrecision, sigFigs, nonSigFigs, groupSeparator, decimalPrecision, ret.sign)
 	if err != nil {
 		return nil, err
 	}
-	ret.core = core
-	tail, err := computeTail(p, v, g, d)
+	ret.stringOut = stringOut
+	//ret.head = computeHead(trunc)
```

Comment:
nit: Delete this inline comment

Reply:
[f5c601c](https://github.com/joshuanario/digits/pull/6/commits/f5c601c1871e252d262cdbe3d035880c5030323c)